lib/iruby/kernel.rb

```diff
@@ -33,8 +36,8 @@ def initialize(config_file)
           hb_socket = c.socket(ZMQ::REP)
           hb_socket.bind(connection % @config['hb_port'])
           ZMQ::Device.new(hb_socket, hb_socket)
-        rescue Exception => ex
-          STDERR.puts "Kernel heartbeat died: #{ex.message}\n"#{ex.backtrace.join("\n")}"
```

Comment:
No this is a thread. Please try it yourself. Create a thread and throw an exception and see what happens.